src/tools/run-make-support/src/lib.rs

```diff
@@ -244,6 +245,28 @@ pub fn uname() -> String {
     output.stdout_utf8()
 }
 
+/// Inside a glob pattern of files (paths), read their contents and count the
+/// number of regex matches with a given expression (re).
+#[track_caller]
+pub fn count_regex_matches_in_file_glob(re: &str, paths: &str) -> usize {
+    let re =
+        regex::Regex::new(re).expect(format!("Regex expression {re} is not valid.").as_str());
+    let paths =
+        glob::glob(paths).expect(format!("Glob expression {paths} is not valid.").as_str());
+    let mut match_count = 0;
+    use io::BufRead;
+    paths.filter_map(|entry| entry.ok()).filter(|path| path.is_file()).for_each(|path| {
+        let file = fs_wrapper::open_file(path);
+        let reader = io::BufReader::new(file);
+        reader
+            .lines()
+            .filter_map(|line| line.ok())
+            .filter(|line| re.is_match(line))
+            .for_each(|_| match_count += 1);
+    });
+    match_count
```

Comment:
I have a feeling this can be streamlined using the `.count()` method to remove the need to manually increment, but I don't know what the most idiomatic style would be

Reply:
```rs
    paths.filter_map(|entry| entry.ok())
        .filter(|entry| entry.as_path().is_file())
        .filter_map(|path| fs::File::open(&path).ok())
        .map(|file| io::BufReader::new(file))
        .flat_map(|reader| reader.lines().filter_map(|entry| entry.ok()))
        .filter(|line| re.is_match(line))
        .count()
```
Good suggestion, after some tinkering I got this to work.